docs/minio-job.md

```diff
@@ -0,0 +1,208 @@
+      env:
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mytestsecret
+              key: PASSWORD
+    - name: add-my-policy
+      op: admin/policy/create
+      args:
+        name: memes-access
+        policy: /temp/policy.json
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
+The `mcImage` field specifies the Docker image that will be used to run the mc commands. In this case, the image is `minio/mc:latest`.
+## serviceAccountName
+The `serviceAccountName` field specifies the name of the Kubernetes ServiceAccount that will be used to run the mc commands. In this case, the ServiceAccount is `mc-job-sa`.
+## securityContext
+example:
+```yaml
+runAsUser: 1000
+runAsGroup: 1000
+fsGroup: 1000
+fsGroupChangePolicy: "OnRootMismatch"
+runAsNonRoot: true
+allowPrivilegeEscalation: false
+capabilities:
+  drop:
+    - ALL
+```
+The `securityContext` field specifies the security context that will be used to run the mc commands. 
+## containerSecurityContext
+The `containerSecurityContext` field specifies the security context that will be used to run the `mc` commands in the container.
+## tenant
+```yaml
+name: tenantName
+namespace: tenantNamespace
+```
+The target tenant that the job will run against.
+## commands
+### args
+if you set this field, the `mc` command will be executed with the arguments.
+`op` must be one of these:
+`make-bucket`, `admin/user/add`,`admin/policy/create`,`admin/policy/attach`, `admin/config/set`, `support/callhome`,`license/register`
+```yaml
+op: make-bucket
+args:
+  name: memes
+```
+Will do a job like `mc mb myminio/memes`
+```yaml
+name: add-my-policy
+op: admin/policy/create
+args:
+  name: memes-access
+policy: /temp/policy.json
+volumeMounts:
+- name: policy
+  mountPath: /temp
+volumes:
+- name: policy
+  configMap:
+    name: mytestconfig
+    items:
+      - key: policy.json
+        path: policy.json
+```
+Will do a job like `mc admin policy create myminio memes-access /temp/policy.json`
+### command
+The `command` field specifies the command that will be executed by the `mc` command.
+`args` must be empty. And `op` can be set to the main command name.
+```
```

Comment:
Is this for arbitrary commands? Just an array list of strings to pass into a constructor, so I could do

```
- op: config
- command:
   - "mc"
   - "admin"
   - "config"
   - "set"
   - "config_setting"
   - "config_param=foo"
   - "config_param=bar"
```

Is this ballpark?

Reply:
I would still like to remove the `op` to be set, because it's not used. Or use `op: command` and use the `args` as the command arguments. I think the current syntax is confusing.

Reply:
Agree with ramon - it's sufficient to support `name` since that is what you would use `dependsOn` with. `op` is fully redundant.

Reply:
As a note - how is this meaningfully different from the pre-baked paths?

I could also do

```
- op: mb
- command:
  - "mc"
  - "mb"
  - "--with-locks"
  - "myminio"
  - "foobar"
```

Right?

Also - assuming `myminio` is a preconfigured alias for the Tenant. Otherwise we have to specify what alias to use here.